### ISSUE TITLE
280: Export doesn't take into account modifiers

### ISSUE THREAD

**Issue by tngreene** [X-Plane/XPlane2Blender#459]
While updating the Blender API usage, there was an issue with xplane_mesh.py, line 49

`mesh = xplaneObject.blenderObject.to_mesh(bpy.context.scene, True)`

After some experimenting and research, I've looked into using something like this

```
import bpy

C = bpy.context
D = bpy.data

modified_mesh = C.object.to_mesh()#preserve_all_data_layers=True, depsgraph=C)
print(len(modified_mesh.vertices)) # 8 verticies

depsgraph = C.evaluated_depsgraph_get()
modified_mesh = depsgraph.id_eval_get(C.object).data
print(len(modified_mesh.vertices)) # 144 verticies
```
from an example of a cube with an array modifier of 18 cubes.

The Blender Manual for Depsgraph shows lots of examples that we could also take from:
https://docs.blender.org/api/current/bpy.types.Depsgraph.html?highlight=depsgraph

My main question in choosing this is where to get depsgraph from, what else we could be missing out on by not getting the id_eval version, and if some of our matrix issues code could be simplified by asking Blender for its evaluated matrix rather than needing to figure it out ourselves. Probably not.

**Comment by tngreene** [comment]
From Kurk of Blender Chat - #python
```
dg = bpy.context.evaluated_depsgraph_get()  
eval_obj = bpy.context.object.evaluated_get(dg)

eval_mesh = eval_obj.to_mesh() 
```

I just want to know what is different between the two of them

**Comment by tngreene** [comment]
Fixed with c0d4205675b!
